protocol/xmrmaker/swap_state.go

```diff
@@ -165,6 +167,98 @@ func newSwapStateFromStart(
 	return s, nil
 }
 
+func checkIfCompletedSuccessfully(
```

Comment:
Can this 90 line function be broken up at all? The name says "check", implying that is is read-only, but if you scroll down enough, it is also updating the state of the swap.

Reply:
yeah, I broke it into two, one that just checks if the swap is already completed (funds claimed) and one that updates the state